cpp/open3d/t/io/file_format/FileSPLAT.cpp

```diff
@@ -0,0 +1,368 @@
+        return false;
+    }
+
+    return true;
+}
+
+bool ReadPointCloudFromSPLAT(const std::string &filename,
+                           geometry::PointCloud &pointcloud,
+                           const open3d::io::ReadPointCloudOption &params) {
+    try {
+
+        //Open the file
+        utility::filesystem::CFile file;
+        if (!file.Open(filename, "r")) {
+            utility::LogWarning("Read SPLAT failed: unable to open file: {}", filename);
+            return false;
+        }
+        pointcloud.Clear();
+
+        // Report progress
+        utility::CountingProgressReporter reporter(params.update_progress);
+        reporter.SetTotal(file.GetFileSize());
+
+        //Constants
+        constexpr size_t data_size = 32;
+        char buffer[data_size];
+        int numberOfPoints = 0;
+
+        // Vectors to store the data
+        std::vector<float> positionVector;
+        std::vector<float> scaleVector;
+        std::vector<float> rotationVector;
+        std::vector<float> opacityVector;
+        std::vector<float> f_dcVector;
+        std::vector<float> f_restVector;
+
+        // Read the data
+        while (file.ReadData(buffer, data_size)) {
+            float positions[3];
+            float scale[3];
+            uint8_t color[4];
+            uint8_t rotation[4];
+
+            // Copy the data into the vectors
+            std::memcpy(positions, buffer, 3 * sizeof(float));
+            std::memcpy(scale, buffer + 3 * sizeof(float), 3 * sizeof(float));
+            std::memcpy(color, buffer + 6 * sizeof(float), 4 * sizeof(uint8_t));
+            std::memcpy(rotation, buffer + (6 * sizeof(float)) + (4 * sizeof(uint8_t)), 4 * sizeof(uint8_t));
+
+            // Calculate the f_dc and f_rest
+            float f_dc[3];
+            float f_rest[3];
+            for (int i = 0; i < 3; i++) {
+                f_dc[i] = ((color[i] / 255.0f) - 0.5) / SH_C0;
+                f_rest[i] = SH_C0;
+            }
+
+            // Calculate the opacity
+            float opacity[1];
+            if (color[3] == 0) {
+                opacity[0] = 0.0f; // Handle division by zero
+            } else {
+                opacity[0] = -log(1 / (color[3] / 255.0f) - 1);
+            }
+            float rot_float[4];
+            for (int i = 0; i < 4; i++) {
+                rot_float[i] = (rotation[i] / 128.0f) - 1.0f;
+            }
+
+            // Add the data to the vectors
+            positionVector.insert(positionVector.end(), positions, positions + 3);
+            scaleVector.insert(scaleVector.end(), scale, scale + 3);
+            rotationVector.insert(rotationVector.end(), rot_float, rot_float + 4);
+            opacityVector.insert(opacityVector.end(), opacity, opacity + 1);
+            f_dcVector.insert(f_dcVector.end(), f_dc, f_dc + 3);
+            f_restVector.insert(f_restVector.end(), f_rest, f_rest + 3);
+
+            reporter.Update(file.CurPos());
```

Comment:
They like to update progress every 1000 iterations

Use a counter:

if (++i % 1000 == 0) {
    reporter.Update(file.CurPos());
}